enhancements/microshift/etcd-as-a-transient-systemd-unit.md

```diff
@@ -0,0 +1,208 @@
+---
+title: etcd-as-a-transient-systemd-unit
+authors:
+  - dusk125
+  - hasbro17
+reviewers:
+  - "@tjungblu, etcd Team"
+  - "@Elbehery, etcd Team"
+  - "@fzdarsky"
+  - "@deads2k"
+  - "@derekwaynecarr"
+  - "@mangelajo"
+  - "@pmtk"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2022-09-16
+last-updated: 2022-10-19
+tracking-link:
+  - https://issues.redhat.com/browse/ETCD-318
+#see-also:
+#  - "/enhancements/this-other-neat-thing.md"
+---
+
+# Decoupling etcd from the MicroShift binary
+
+## Summary
+
+The enhancement proposes moving etcd out of the MicroShift binary and into a transient systemd unit that is launched and managed by MicroShift.
+
+## Motivation
+
+Currently MicroShift [bundles upstream etcd](https://github.com/openshift/microshift/blob/f7f2260e4fbff61654f478fc149cb7052261f87f/go.mod#L21) into its binary and runs it as a goroutine.
+This results in etcd dependencies being coupled with MicroShift which makes it harder to maintain and upgrade etcd independently of the platform like it is done with a separate openshift/etcd repo in the Openshift Container Platform (OCP).
+
+This enhancement outlines an update to MicroShift to change how etcd is bundled and deployed with MicroShift in order to minimize risk of shared dependencies between etcd and OpenShift/Kubernetes leading to hard-to-trace bugs.
+
+Changing the deployment of etcd, outlined below, would alleviate concerns about building etcd with a version of shared dependencies - mainly grpc - that etcd is not currently built against, and therefore, not tested against.
+
+Additionally moving the etcd deployment into its own process would make it easier to debug and process logs via journatlctl.
+
+### User Stories
+
+1. "As an etcd developer, I want to maintain etcd versions independently of microshift dependencies."
+
+2. "As a MicroShift Device Administrator, I want to observe and debug the etcd server logs separately from other microshift processes."
+
+### Goals
+
+- The etcd binary is built and shipped separately from the microshift binary to ease maintainence.
```

Comment:
Sorry for late comment, but in this step we won't ship etcd binary separately from microshift's binary, right? They will be in the same rpm, installed to the same location.
```suggestion
- The etcd is built as a separate binary to ease dependency management and shipped together with MicroShift binary in single RPM to ease deployment process.
```

Reply:
Yes that's much more clear, thanks!